daemon/start.go

```diff
@@ -189,21 +189,26 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 			}
 			err = daemon.containerd.Create(ctx, container.ID, spec, shim, createOptions)
 		}
-		if err != nil {
-			return translateContainerdStartErr(container.SetExitCode, err)
+		if exitCode, err := translateContainerdStartErr(err); err != nil {
+			if exitCode != -1 {
```

Comment:
Maybe we should make a constant for this "magic" number?

Reply:
Heh; so I initially creating `const's` for the possible exit-codes, but the descriptions became a bit "fuzzy" (especially the `127` being "could be this, or could be that"), so I removed them, but yeah, perhaps I should have some consts (it's not exported, so I guess they don't have to be squeaky clean on description.

Reply:
Yeah, I think this is the only one that stuck out because it's the only one I see special parsing for, likely because it's the only one that's new? :innocent:

Reply:
I pushed a commit with const/enums (but I feel like it's a bit too much perhaps). Alternatively, I could change the checks to `if exitCode > 0` do you think that would make it clearer?

Reply:
Actually, perhaps I should just make sure that it always returns a sensible exit-code. I wasn't sure if the exitCode could be set through other means, but looking at both `state.SetExitCode()`, and `state.ExitCodeValue`, then it looks like all updates happen within this file:

<img width="898" alt="Screenshot 2022-02-16 at 20 12 33" src="https://user-images.githubusercontent.com/1804568/154339645-2b42fa6f-79f3-4c23-bff7-611a33d7fc57.png">

<img width="595" alt="Screenshot 2022-02-16 at 20 17 29" src="https://user-images.githubusercontent.com/1804568/154339650-a27d57b2-a7f0-4d3b-93fd-9e1709ca487c.png">


And if it's not set, it's updated to `128` in the `defer()` https://github.com/moby/moby/blob/9674540ccff358c3cd84cc2f33c3503e0dab7fb7/daemon/start.go#L124-L127

From that, I'm tempted to say that we can return `128` as a default (for "unknown error")

Reply:
Arf.. I may be looking at the wrong `SetExitcode()` (state vs config?); let me get back on you on that

Reply:
Pushed a commit with that change; I can squash some of the commits